plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt

```diff
@@ -307,14 +350,34 @@ class PluginService(
         }
     }
 
-    private fun findPluginConfiguration(pluginDefinitionKey: String, filter: (JsonNode) -> Boolean): PluginConfiguration? {
+    private fun findPluginConfiguration(
+        pluginDefinitionKey: String,
+        filter: (JsonNode) -> Boolean
+    ): PluginConfiguration? {
         val configurations = pluginConfigurationRepository.findByPluginDefinitionKey(pluginDefinitionKey)
         return configurations.firstOrNull { config ->
             val configProperties = config.properties
             configProperties != null && filter(configProperties)
         }
     }
 
+    private fun PluginConfiguration.runAllPluginEvents(eventType: PluginEvent.EventType): PluginConfiguration {
```

Comment:
The overload function can just be a service method

Reply:
Yes, it can be a normal method, but that removes the ability to chain it. Having it be an extension method for the PluginConfiguration class is purely a qol choice.
That said, if you are against this, then I will rework it.